[Ruby] Concept Exercise: arrays (#1611)
exercism/v3: languages/ruby/exercises/concept/arrays/.docs/introduction.md

> In Ruby, **arrays** are ordered, integer-indexed collections of any object. Array indexing starts at 0, as in C or Java. A negative index is assumed to be relative to the end of the array—that is, an index of -1 indicates the last element of the array, -2 is the next to last element in the array, and so on.

Reviewer: ```suggestion
In Ruby, **arrays** are ordered, integer-indexed collections of any object. Array indexing starts at `0`. A negative index is assumed to be relative to the end of the array — i.e. an index of `-1` indicates the last element of the array, `-2` is the next to last element in the array, and so on.
```

> Ruby arrays mixed in the [Enumerable module][enumerable-module] which provides collection classes with several traversal and searching methods, and with the ability to sort.

Reviewer: ```suggestion
Ruby arrays mix in the [Enumerable module][enumerable-module], which adds several traversal and searching methods, and with the ability to sort.
```

### Create array.

- An array in Ruby can contain different types of objects.

```ruby
array = [1, "two", 3.0] #=> [1, "two", 3.0]
```

> - An array can also be created by explicitly calling ::new

Reviewer: ```suggestion
- An array can also be created by explicitly calling `::new`
```

> - An array can also be created by explicitly calling ::new
> 
> ```ruby
> array = Array.new #=> []
> Array.new(3)      #=> [nil, nil, nil]
> ```

Reviewer: ```suggestion
```

I suggest leaving this out as it's not needed in the exercise.

### Element Assignment

> - Sets the element at index, or replaces a subarray from the start index for length elements.

Reviewer: ```suggestion
Elements can accessed or changed using indexes. Subarrays can be accessed by specifying a start index and a size.
```

```ruby

> a = Array.new

Reviewer: ```suggestion
a = ["", "", "", "", ""]
```

Reviewer: Let's not add the Array.new syntax for now, as we haven't taught `nil` anywhere, and we probably don't want to add that as a prereq just for this example.

> a[4] = "4"                  #=> [nil, nil, nil, nil, "4"]

Reviewer: ```suggestion
a[4] = "hello"  #=> [nil, nil, nil, nil, "hello"]
```

Reviewer: I think it's clearer to use a string rather than a number in a string here.

> a[0, 3] = [ 'a', 'b', 'c' ] #=> ["a", "b", "c", nil, "4"]

Reviewer: ```suggestion
a[0, 3] = [ 'a', 'b', 'c' ] #=> ["a", "b", "c", nil, "hello"]
```

```

- Negative indices will count backward from the end of the array.

```ruby

> a = ['a, 'b']

Reviewer: ```suggestion
a = ['a', 'b']
```

> a[-1] = "Z" #=> ["a", "Z"]

Reviewer: ```suggestion
a[-1] = "Z" 
a #=> ["a", "Z"]
```

The first line actually returns `"Z"` so we should have a second line printing the array.

> 

Reviewer: ```suggestion
```

```

> ### Element Reference

Reviewer: ```suggestion
### Element Reference
```

> In Ruby, **arrays** are ordered, integer-indexed collections of any object. Array indexing starts at `0`. A negative index is assumed to be relative to the end of the array — i.e. an index of `-1` indicates the last element of the array, `-2` is the next to last element in the array, and so on.

Reviewer 2: This introduction file is on the large side. Maybe it would be possible to trim things done slightly? The best way to go about this is to:

1. Remove all knowledge that the student does not need to know to be able to solve the exercise
1. Combine examples
